docs/manual/technical_design.md

```diff
@@ -0,0 +1,334 @@
+
+The locking system is used to protect operations that have global scope or otherwise have a global impact. This includes operations like setting database configuration, which impacts the entire cluster. It also includes operations that trigger recoveries or that we want to restrict to one DC at a time, such as excluding processes.
+
+Because this locking system involves writing to the database, it will not work when the database is unavailable. In that situation any attempt to aquire a lock will fail. If the database is unavailable and you need the operator to take action to make it available, you can work around this by setting the `disableLocks` field in the lock options to `true`. However, many of the actions that require locks are activities that are impossible or unsafe when the database is unavailable, and often an unavailable database will require manual intervention.
+
+If there is a dysfunctional instance of the operator that cannot be trusted to perform global operations, you can block it from taking locks by adding its `lockID` to the deny list in the cluster spec. You can set this value in any DC. This will only affect operations on the cluster whose spec you update. This will set the key `\xff\x02/org.foundationdb.kubernetes-operator/denyList/$lockID` to the the value `$lockID`. If an instance of the operator with that lock ID sees that the key is set, it will fail any attempt to acquire a lock, even if it has a lock already. Any other instance of the operator that sees an active lock for an instance in the deny list will ignore that lock and will be able to take one for itself.
+
+In order to avoid contention between different instances of the operator in managing the deny list, if the operator has no entry in the deny list in its spec for a given `lockID`, it will take no action. This means that if you set the deny list in the spec, and then clear that field in the spec, the deny list will still be set in the database. In order to effectively remove an entry from the deny list, you have to update its entry with the flag `allow: true`. This tells the operator that your intention is to explicitly allow this instance to take global locks again.
+
+As an example, you would use this spec to add the operator in `dc1` to the deny list:
+
+```yaml
+spec:
+    lockOptions:
+        denyList:
+            - id: dc1
+```
+
+And you would use this spec to remove the operator in `dc1` from the deny list:
+
+```yaml
+spec:
+    lockOptions:
+        denyList:
+            - id: dc1
+              allow: true
+```
+
+Once that is done and the change is reconciled, you can clear the deny list in the spec.
+
+See the [LockOptions](../cluster_spec.md#LockOptions) documentation for more options for customizing the locking system.
+
+## Cluster Reconciliation
+
+The cluster reconciler runs the following subreconcilers:
```

Comment:
Should we rather point to the code here?

Reply:
I think it's good to have the list of subreconcilers embedded in this doc, to give a sense of the structure for the subsections below.